crates/astria-conductor/src/executor/mod.rs

```diff
@@ -728,7 +728,7 @@ fn does_block_response_fulfill_contract(
         ExecutionKind::Firm => state.firm_number(),
         ExecutionKind::Soft => state.soft_number(),
     };
-    let expected = current + 1;
+    let expected = current.saturating_add(1);
```

Comment:
I'm not sure how we handle assigning a height when creating a new block if the latest is at height `u32::MAX`.  So `saturating_add` might be the wrong call here?  It's possibly better to do `checked_add().unwrap()` if we don't expect to ever reach that height?

Reply:
IMO add an a variant to `ContractViolation` (i.e. make it an enum) for this case, so `current.checked_add(1).ok_or(...)`

Reply:
Done in [44c99af](https://github.com/astriaorg/astria/pull/1081/commits/44c99affc52d66ee3134fa1bbc23f66d21bb632a).